Mathlib/Topology/Algebra/LinearTopology.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández, Anatole Dedecker
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Operations
+import Mathlib.Topology.Algebra.Ring.Basic
+import Mathlib.Topology.Algebra.OpenSubgroup
+
+/-! # Linear topologies on rings
+
+Following Bourbaki, *Algebra II*, chapter 4, §2, n° 3, a topology on a ring `R` is *linear* if
+it is invariant by translation and admits a basis of neighborhoods of 0 consisting of
```

Comment:
Does this imply that it is a topological ring: i.e. multiplication as bilinear map is continuous?

Reply:
Yes, but Bourbaki does it the other way round, I believe. In mathlib, we have to start with imposing that we have a topological ring. This is a bit sad, because for `IsLinearTopology.mk_of_twoSidedIdeal`, we could (mathematically) dispose of the hypothesis that we start from a topological ring.

Reply:
Maybe we could bundle the topological ring hypothesis in the class `IsLinearTopology`.

Reply:
@ADedecker , what do you think?